### PR TITLE
prevent the merge of two vertices that would create a non-manifold edge

### PR DESCRIPTION
**Pull request by sloriot**
Without this fix the stitching can create non-manifold edges by merging two vertices that are already incident to the same vertex. This PR avoid doing this.

### CODE REVIEW

File: Polygon_mesh_processing/test/Polygon_mesh_processing/test_stitching.cpp [hunk 1]
```diff
@@ -78,6 +79,7 @@ int main()
   test_surface_mesh("data_stitching/multiple_incidence.off");
   test_surface_mesh("data_stitching/incidence_3.off");
   test_polyhedron<Epic>("data_stitching/incoherent_patch_orientation.off");
+  test_polyhedron<Epic>("data_stitching/non_stitchable.off");
```
Comment (lrineau) [inline Polygon_mesh_processing/test/Polygon_mesh_processing/test_stitching.cpp]: Why add the same call twice?

Comment (lrineau) [inline Polygon_mesh_processing/test/Polygon_mesh_processing/test_stitching.cpp]: Same for the other line, that is also twice:
```c++
test_polyhedron<Epic>("data_stitching/incoherent_patch_orientation.off");
```

**Comment by sloriot** [comment]
The patch is not sufficient, we must check that the intersection of the link of the two vertices to be merged is empty (and not only considering the first incident edge)

**Comment by sloriot** [comment]
I fixed the test.

**Pull request merged**